src/platforms/dotnet/guides/aspnetcore/performance/included-instrumentation.mdx

```diff
@@ -0,0 +1,137 @@
+
+</Note>
+
+### Server Integration
+
+ASP.NET Core integration, once enabled, captures each incoming HTTP request and turns it into a transaction with the help of a custom middleware. To enable tracing on ASP.NET Core, you need to update your pipeline configuration to include a call to `UseSentryTracing()`:
+
+```csharp
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Sentry.AspNetCore;
+
+public class Startup
+{
+    /* ... */
+
+    public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+    {
+        if (env.IsDevelopment())
+        {
+            app.UseDeveloperExceptionPage();
+        }
+        else
+        {
+            app.UseExceptionHandler("/Home/Error");
+            app.UseHsts();
+        }
+
+        app.UseHttpsRedirection();
+        app.UseStaticFiles();
+
+        app.UseRouting();
+
+        // Enable automatic tracing integration.
+        // Make sure to put this middleware right after `UseRouting()`.
+        app.UseSentryTracing();
+
+        app.UseAuthorization();
+
+        app.UseEndpoints(endpoints =>
+        {
+            endpoints.MapControllerRoute(
+                name: "default",
+                pattern: "{controller=Home}/{action=Index}/{id?}");
+        });
+    }
+}
+```
+
+Transaction names follow the pattern `<HTTP method> <Route>`; for example, a request to the following action will create a transaction named `GET /person/{id}`:
+
+```csharp
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+public class HomeController : Controller
+{
+    [HttpGet("/person/{id}")]
+    public IActionResult Person(string id) { /* ... */ }
+}
+```
+
+Transactions created by this integration are automatically added to the current scope. If you want to start your own spans from the automatically created transaction, you can use the following approach:
+
+```csharp
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Sentry;
+
+public class HomeController : Controller
+{
+    private readonly IHub _sentryHub;
+
+    public HomeController(IHub sentryHub) => _sentryHub = sentryHub;
+
+    [HttpGet("/person/{id}")]
+    public IActionResult Person(string id)
+    {
+        // Get the last non-finished span (or transaction).
+        // This can generally be null, but since the root transaction
+        // is set automatically by the middleware, we can expect it
+        // to always be defined.
+        var currentSpan = hub.GetSpan();
+
+        // We can then start our own spans which will be attached to
+        // the automatically created transaction.
+        var childSpan = currentSpan?.StartChild("additional-work");
+
+        // ...
+
+        childSpan?.Finish();
+
+        // Note: don't call `currentSpan.Finish()` here as it may prematurely
```

Comment:
This is surprising to me to be honest. Maybe we shouldn't make the transctio available through `hub.GetSpan()` then?

Should we just have `hub.StartChild` then?

Reply:
Perhaps. I raised this concern earlier when we were discussing adding `GetSpan()`.
It can have even more severe conflicts if you consider the HTTP client factory filter that starts spans off of HTTP requests.

Reply:
At least for the docs lets remove that for now, keeping just StartChild.
We need to revisit this Transaction API stuff anyway

Reply:
Do you mean remove the comment or the entire section explaining `GetSpan()`?

Reply:
Yeah remove `GetSpan`. Just leave `StartChild()` so they can create child spans

Reply:
I'm confused, are we talking about the docs or the library API?
There is no `StartChild()` in the API, only `GetSpan()`.

Reply:
Let's review the api before we merge this